src/services/rfqm_service.ts

```diff
@@ -711,34 +724,61 @@ export class RfqmService {
         workerAddress: string,
         callData: string,
     ): Promise<SubmissionsMap> {
-        // make sure this order hasn't been used by another worker (or this worker before)
-        if ((await this._dbUtils.findRfqmTransactionSubmissionsByOrderHashAsync(orderHash)).length !== 0) {
-            throw new Error('found pre-existing transaction submissions');
-        }
-
         let submissionsMap: SubmissionsMap = {};
 
         let gasPrice: BigNumber;
-        let nonce: number;
         let gasEstimate: number;
+        let nonce: number;
 
-        [gasPrice, nonce, gasEstimate] = await Promise.all([
-            this._protocolFeeUtils.getGasPriceEstimationOrThrowAsync(),
-            this._blockchainUtils.getNonceAsync(workerAddress),
-            this._blockchainUtils.estimateGasForExchangeProxyCallAsync(callData, workerAddress),
-        ]);
+        const previousSubmissions = await this._dbUtils.findRfqmTransactionSubmissionsByOrderHashAsync(orderHash);
+
+        // if there are previous tx submissions (repair mode), pick up monitoring and re-submission
+        if (previousSubmissions.length !== 0) {
+            // setting values to override them
+            nonce = -1;
+            gasPrice = new BigNumber(0);
+            for (const submission of previousSubmissions) {
+                gasPrice = new BigNumber(0);
+                // make sure this order hasn't been submitted by another worker
+                if (submission.from !== workerAddress) {
+                    throw new Error('found tx submissions from a different worker');
+                }
+                submissionsMap[submission.transactionHash!] = submission;
 
-        const expectedTakerTokenFillAmount = this._blockchainUtils.getTakerTokenFillAmountFromMetaTxCallData(callData);
+                if (nonce === -1) {
+                    nonce = submission.nonce!;
+                } else {
+                    if (submission.nonce! !== nonce) {
+                        throw new Error('found different nonces in tx submissions');
+                    }
+                }
+                if (submission.gasPrice!.gt(gasPrice)) {
+                    gasPrice = submission.gasPrice!;
+                }
+            }
+            gasEstimate = await this._blockchainUtils.estimateGasForExchangeProxyCallAsync(callData, workerAddress);
```

Comment:
hmm, just experimenting around here, but what if we introduced two helper methods `recoverSubmissionContext` and `initializeSubmissionContext` like so:
 
```
function _recoverPreviousSubmissionContext(previousSubmissions): [gasPrice, nonce, gasEstimate, SubmissionsMap]
``` 

and:

```
function _initializeSubmissionContext(orderHash, workerAddress, callData): [gasPrice, nonce, gasEstimate, SubmissionsMap]
``` 
And then usage could look like this:

```
const [gasPrice, nonce, gasEstimate, submissionsMap] =  previousSubmissions.length > 0 
  ? this._recoverPreviousSubmissionContext(previousSubmissions)
  : this._initializeSubmissionContext(orderHash, workerAddress, callData);
```

Helps with the nesting and keeping this function tidy. Thoughts?